doc/sphinx/source/community/backward_compatibility.rst

```diff
@@ -0,0 +1,263 @@
+selection (e.g. important new, scheduled for removal, and removed
+features) also by other means, to ensure we reach as many people
+potentially affected as possible (see “Guidance on handling
+*backward-incompatible changes*” section below).
+We organize monthly community meetings where *recipe developers* can learn about
+the latest developments and everyone is welcome to join, ask questions, and
+provide feedback.
+
+To meet the needs of users and funders, we should take reproducibility
+of older results seriously, but this should not hold us back from
+advancing our tools. We can support this by uploading a well tested
+container image to an archive that provides a DOI and by providing clear
+instructions on how to use such containers.
+
+Helping developers to upgrade
+-----------------------------
+
+*Recipe users* of ESMValTool will be able to successfully run
+*integrated recipes* using a *release*, since all
+*backwards-incompatible changes* introduced between *releases* will have
+been fixed before the *release* is created. Please note the
+:ref:`broken recipe policy<broken-recipe-policy>`.
+
+However, *recipe developers* working on *user recipes* must be provided
+with information to enable them to adapt their code to resolve issues
+related to *backwards-incompatible changes* when *backwards-incompatible
+changes* are introduced to the main branch / when a *release* of
+ESMValTool is created.
+
+Guidance on handling *backward-incompatible changes*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As well as helping users to handle *backward-incompatible changes*, the
+policy and surrounding tools must help developers avoid making
+*backwards-incompatible changes*. Not many ideas developed on this yet,
+but components should include:
+
+-  Testing; *backwards-incompatible changes* should be discovered as
+   early in the development process as possible. This motivates
+   continued investment in automated testing.
+-  Guidance on how to minimise the likelihood of introducing
+   *backward-incompatible changes* and how to use deprecation warnings
+   when needed (see `developer
+   guidance <https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility>`__).
+-  Instructions on how to provide the information that *recipe
+   developers* need to adapt to the *backward-incompatible change*,
+   e.g. the text that will go into *release* notes if we follow the
+   proposal above.
+-  General instructions for *recipe developers* working on *user
+   recipes* to enable them to adapt their code related to
+   *backwards-incompatible changes* (see `ESMValTool_Tutorial: issue
+   #263 <https://github.com/ESMValGroup/ESMValTool_Tutorial/issues/263>`__).
+-  The developer or reviewer must tag the core development team to
+   notify them of the *backward-incompatible change*, and give at least
+   2 weeks for objections to be raised before merging to the main
```

Comment:
This also needs to be linked and or mentioned in the following places in the documentation:
- https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#backward-compatibility
- https://esmvaltool--2879.org.readthedocs.build/en/2879/community/review.html#merge

Reply:
Opened an ESMValCore PR (ESMValGroup/ESMValCore/#1916) for the first one (assuming that an ESMValCore branch requires its own PR)

Reply:
Added references in both of the above: https://github.com/ESMValGroup/ESMValCore/commit/8af608a679866a2ce479f942e223f872ef66ab08 and https://github.com/ESMValGroup/ESMValTool/commit/14b482df3dddeb0355cc2b78bb617b4b4193f2e5. The first one in particular should be reviewed, as the release notes instructions previously said nothing about a section on backward incompatible changes, and assumed that the release notes will be automated. My change to the instructions introduces manual editing of the ESMValCore release notes.

Reply:
I added a bit of text along those lines. Approving should be fine because then the author of the pull request knows they are done and can start working on something new, unless anyone objects later.